articles/openshift/oshift-how-disable-weak-cipher-suites-enable-hsts.md

```diff
@@ -0,0 +1,102 @@
+## Overview
+
+This article outlines steps to increase the security of externally exposed routes from your OpenShift cluster, through the use of more secure cipher suites and HSTS.
+
+### Intended audience
+
+OpenShift developers who have created and deployed services into OpenShift, and created and exposed secure routes for those services.
+
+## Disable weak cipher suites
+
+Cipher suites perform an important role in determining the security of a TLS connection. Over time vulnerabilities have been identified in the algorithms used by cipher suites which has resulted in published methods of decrypting and manipulating encrypted traffic. It is therefore generally recommended to use as modern a cipher suite as possible.
+
+### UKCloud for OpenShift v3.x
+
+#### Data plane
+
+In order to provide extended client compatibility, the OpenShift router supports cipher suites from TLS 1.0 - TLS 1.2 by default. Some of these cipher suites (especially older ones) are vulnerable to man-in-the-middle (MITM) attacks.
+
+If exposed secure routes are only being accessed from modern clients (that support TLS 1.2 and above) the weak cipher suites can be disabled by adding the following environment variable to any router `DeploymentConfig`s:
+
+```
+oc set env dc/router ROUTER_CIPHERS=modern -n default
+```
+
+This will trigger a rollout of the `DeploymentConfig`, going forward the new router pods will only accept cipher suites from TLS 1.2.
+
+#### Control plane
+
+By default, the OpenShift control plane (web console) only supports TLS 1.2 ciphers, although these do not provide forward secrecy.
+
+It is possible to restrict the ciphers to only ones that provide forward secrecy, to do so please raise a Service Request to the UKCloud OpenShift Support team via the [My Calls](https://portal.skyscapecloud.com/support/ivanti) section of the UKCloud Portal.
+
+### UKCloud for OpenShift v4.x
+
+#### Data plane
+
+There are no weak ciphers supported by the OpenShift `IngressController` since by default it supports TLS 1.2 cipher suites with forward secrecy. It is possible to further restrict the ciphers by modifying the default `IngressController` object, specifically the value of `IngressController.spec.tlsSecurityProfile`
+
+```
+oc edit ingresscontroller default -n openshift-ingress-operator
+```
+
+#### Control plane
+
+There are no weak ciphers supported by the OpenShift control plane (API) since by default it supports TLS 1.2 and 1.3 cipher suites with forward secrecy.
+
+## Enable HSTS
+
+HSTS (HTTP Strict Transport Security) prevents MITM attacks such as protocol downgrading by setting a header within HTTPS responses. This header (`Strict-Transport-Security`) is cached by a browser and ensures that future requests to the same host always use HTTPS as opposed to HTTP.
+
+> [!NOTE]
+> It is currently not possible to enable HSTS for the OpenShift v3.x web console or the OpenShift v4.x API server.
+
+HSTS can be enabled on secure routes by using the following command:
```

Comment:
We should include a warning that says to only do this on secure routes that have valid certificates. I guess all prod clusters will produce domain names which are valid with the default domains for routes, but we'd need to explain that secure custom-domain-name routes need a valid cert against the route?

Reply:
@gellner just pushed an update containing a warning